searx/engines/demo_online.py

```diff
@@ -0,0 +1,92 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""Within this module we implement a *demo online engine*.  Do not look to
+close to the implementation, its just a simple example which queries `The Art
+Institute of Chicago <https://www.artic.edu>`_
+
+To get in use of this *demo* engine add the following entry to your engines
+list in ``settings.yml``:
+
+.. code:: yaml
+
+  - name: my online engine
+    engine: demo_online
+    shortcut: demo
+    disabled: false
+
+"""
+
+from json import loads
+from urllib.parse import urlencode
+
+engine_type = 'offline'
+categories = ['general']
+disabled = True
+timeout = 2.0
+categories = ['images']
+paging = True
+page_size = 20
+
+search_api = 'https://api.artic.edu/api/v1/artworks/search?'
+image_api = 'https://www.artic.edu/iiif/2/'
+
+about = {
+    "website": 'https://www.artic.edu',
+    "wikidata_id": 'Q239303',
+    "official_api_documentation": 'http://api.artic.edu/docs/',
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": 'JSON',
+}
+
+
+# if there is a need for globals, use a leading underline
+_my_online_engine = None
+
+def init(engine_settings):
+    """Initialization of the (online) engine.  If no initialization is needed, drop
+    this init function.
+
+    """
+    global _my_online_engine  # pylint: disable=global-statement
+    _my_online_engine = engine_settings.get('name')
```

Comment:
Hm, I guess I have a problem to understand why we have this argument `engine_settings` .. does it make sense to drop this argument? .. for me it was a bit confusing that we update module's name space with the values from YAML (`engine_data`):

https://github.com/searxng/searxng/blob/5f76238d5cd6f9c850d5237fe639ba2900fd3e6c/searx/engines/__init__.py#L117-L125

and later, when the processor is initialized ...

https://github.com/searxng/searxng/blob/5c5db719d2039d34aa1426bf7eb3e57055d4a54a/searx/search/processors/abstract.py#L70-L72

we call a function that gather `engine_data` from the YAML settings.

In sense of _single point of definition_ I guess it is better to say, that everything is in the namespace of engine's modul and this can be configured by (is updated from) the YAML file.

But may be I'm totally wrong, since I do not have a complete overview.